x/monitoringc/types/genesis.go

```diff
@@ -28,47 +28,73 @@ func (gs GenesisState) Validate() error {
 	if err := host.PortIdentifierValidator(gs.PortId); err != nil {
 		return err
 	}
+
 	// Check for duplicated index in verifiedClientID
 	verifiedClientIDIndexMap := make(map[string]struct{})
-
+	clientIDMap := make(map[string]struct{})
 	for _, elem := range gs.VerifiedClientIDList {
 		index := string(VerifiedClientIDKey(elem.LaunchID))
 		if _, ok := verifiedClientIDIndexMap[index]; ok {
 			return fmt.Errorf("duplicated index for verifiedClientID")
 		}
 		verifiedClientIDIndexMap[index] = struct{}{}
+
+		// Check for duplicated client id
+		for _, clientID := range elem.ClientIDs {
+			key := clientIDKey(elem.LaunchID, clientID)
+			if _, ok := clientIDMap[key]; ok {
+				return fmt.Errorf("duplicated client id")
+			}
+			clientIDMap[key] = struct{}{}
+		}
 	}
+
 	// Check for duplicated index in providerClientID
 	providerClientIDIndexMap := make(map[string]struct{})
-
 	for _, elem := range gs.ProviderClientIDList {
 		index := string(ProviderClientIDKey(elem.LaunchID))
 		if _, ok := providerClientIDIndexMap[index]; ok {
 			return fmt.Errorf("duplicated index for providerClientID")
 		}
+		// Check if the client id exist
+		key := clientIDKey(elem.LaunchID, elem.ClientID)
+		if _, ok := clientIDMap[key]; !ok {
+			return fmt.Errorf("client id from ProviderClientIDList list not found")
+		}
 		providerClientIDIndexMap[index] = struct{}{}
 	}
+
 	// Check for duplicated index in launchIDFromVerifiedClientID
 	launchIDFromVerifiedClientIDIndexMap := make(map[string]struct{})
-
 	for _, elem := range gs.LaunchIDFromVerifiedClientIDList {
 		index := string(LaunchIDFromVerifiedClientIDKey(elem.ClientID))
 		if _, ok := launchIDFromVerifiedClientIDIndexMap[index]; ok {
 			return fmt.Errorf("duplicated index for launchIDFromVerifiedClientID")
 		}
+		// Check if the client id exist
+		key := clientIDKey(elem.LaunchID, elem.ClientID)
+		if _, ok := clientIDMap[key]; !ok {
+			return fmt.Errorf("client id from launchIDFromVerifiedClientID not found")
+		}
```

Comment:
We should rather check if it is in `providerClientIDIndexMap`

Reply:
Should we check if the `launchID` exists in the `providerClientIDIndexMap`? Or should we fetch all and iterate to check if the `clientID` exist? because the `providerClientIDIndexMap` is indexed by `launchID`

Reply:
My bad, this is correct, there is one entry in `LaunchIDFromVerifiedClientID` per verified client ID